### PR TITLE
Add Health Check Elasticsearch tests

### LINKED ISSUE #3054: Improve PHP Unit Tests Coverage
**Is your enhancement related to a problem? Please describe.**

This bug will be used to improve the overall coverage of PHP Unit Tests

### PR DESCRIPTION
**Pull request by burhandodhy**
### Description of the Change

This PR adds the unit tests for the `HealthCheckElasticsearch` https://github.com/10up/ElasticPress/blob/4.4.0/includes/classes/HealthCheck/HealthCheckElasticsearch.php

Closes #3054

### Changelog Entry

> Added - New unit tests for the `HealthCheckElasticsearch` class.

### Credits

Props @burhandodhy 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: tests/php/TestHealthCheckElasticsearch.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		// Make the request.
+		try {
+			$this->_handleAjax( 'health-check-elasticpress-health-check-elasticsearch' );
+		} catch ( WPAjaxDieContinueException $e ) {
+			unset( $e );
+		}
+
+		$response = json_decode( $this->_last_response, true );
+
+		$this->assertEquals( true, $response['success'] );
```
Comment (felipeelia) [inline tests/php/TestHealthCheckElasticsearch.php]: Can we change this to
```suggestion
		$this->assertTrue( $response['success'] );
```

File: tests/php/TestHealthCheckElasticsearch.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		$this->assertEquals( 'Your site could not connect to Elasticsearch', $response['data']['label'] );
+		$this->assertEquals( 'critical', $response['data']['status'] );
+		$this->assertEquals( 'ElasticPress', $response['data']['badge']['label'] );
+		$this->assertEquals( 'red', $response['data']['badge']['color'] );
+		$this->assertEquals( 'The Elasticsearch host is not set.', $response['data']['description'] );
+
+		remove_filter( 'ep_host', '__return_empty_string' );
```
Comment (felipeelia) [inline tests/php/TestHealthCheckElasticsearch.php]: I _think_ the test suite removes all filters in the end of each test execution (at least WP core does.) Do you mind testing if we really need to remove it @burhandodhy?

**Review by felipeelia — CHANGES_REQUESTED** [review]
Great work on this one @burhandodhy. I left a couple of comments, do you mind taking a look? Thanks!

File: tests/php/TestHealthCheckElasticsearch.php (hunk 2 above)
Comment (burhandodhy) [reply tests/php/TestHealthCheckElasticsearch.php]: The tests are working fine without removing the filters. Most of the tests involve removing the filters at the end. Do you mind if I remove `remove_filters` from all tests in a separate PR?

File: tests/php/TestHealthCheckElasticsearch.php (hunk 1 above)
Comment (burhandodhy) [reply tests/php/TestHealthCheckElasticsearch.php]: Done

**Pull request merged**